pkgs/development/python-modules/rctclient/default.nix

```
@@ -0,0 +1,59 @@
+{
+  lib,
+  #  aiohttp,
+  buildPythonPackage,
+  fetchFromGitHub,
+  #pytest-aiohttp,
+  #pytestCheckHook,
+  pythonOlder,
+  typing,
+  click,
+  scapy,
+  sphinx,
+  sphinx-autodoc-typehints,
+  sphinx-rtd-theme,
+}:
+
+buildPythonPackage rec {
+  pname = "rctclient";
+  version = "0.0.4";
+  format = "setuptools";
+
+  disabled = pythonOlder "3.8";
+
+  src = fetchFromGitHub {
+    owner = "svalouch";
+    repo = "python-rctclient";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-QPla5h8wSM9Ynj44Uwc1a2yAnu8TXbyBWzVHQeW6jnI=";
+
+  };
+
+  postPatch = ''
+    # see comment on https://github.com/cmroche/moonraker-api/commit/e5ca8ab60d2839e150a81182fbe65255d84b4e4e
+    #substituteInPlace setup.py \
+    #  --replace 'name="rctclient",' 'name="rctclient",version="${version}",'
+  '';
+
+  propagatedBuildInputs = [
+    typing
+    click
+    scapy
+    sphinx
+    sphinx-autodoc-typehints
+    sphinx-rtd-theme
+  ];
```

```suggestion
  optional-dependencies.cli = [ click ];
```

This package has no mandatory dependencies. The only optional-dependency we should set up is `cli` with click. Don't bother with docs and tests.

https://github.com/weltenwort/python-rctclient/blob/main/setup.py#L29-L35